### PR TITLE
Wrap class variables

### LINKED ISSUE #1618: Add wrapping of class variables
### Describe the feature
To complete the wrapping of a class it should be possible to access and modify class variables.

### Test Code
Provide code which does not currently work but which should do when this issue is fixed:
```python
class Point:
    def __init__(self : 'Point', x : float):
        self.x = x
```

### Proposed Solution
The wrapping stage should generate setters and getters for the classes. The getters and setters will allow even scalars to be accessed and modified reliably

### PR DESCRIPTION
**Pull request by EmilyBourne**
Add the wrapping of class variables. Getters and setters are created which can be called from Python. These getters and setters are packed into a class called `PyGetSetDefElement` to expose the property. Fixes #1618 .

**Commit Summary**
- Update documentation to remove the reference to incomplete wrapping
- Add `BindCClassProperty` to hold the getters and setters necessary to wrap a class property in Fortran and expose it to C
- Add `PyGetSetDefElement` to hold the getters and setters necessary to wrap a class property in C and expose it to Python
- Update version to 1.11.0
- Remove obsolete xfail mark

### CODE REVIEW

**Comment by EmilyBourne** [comment]
Blocked by #1715 so that the 2 can share the implementation of `_incref_return_pointer`

**Comment by EmilyBourne** [comment]
/bot run docs spelling pylint pyccel_lint coverage

File: pyccel/ast/bind_c.py [hunk 1]
```diff
@@ -572,6 +573,90 @@ def wrapper_function(self):
...
+        return self._python_name
+
+    @property
+    def docstring(self):
+        """
+        The docstring of the property being wrapped.
+
+        The docstring of the property being wrapped.
+        """
+        return self._docstring
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/bind_c.py:656]: /bot accept
Attributes don't have docstrings. This code will be useful when the same class is used to handle properties

**Review by Farouk-Echaref — APPROVED** [review]
Looks good to me, tests describe well what this feature supports

**Review by yguclu — APPROVED** [review]
Looks good!

**Pull request merged**